File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string =
    let p (modulus, str) = if 0 = number % modulus then str else ""
    [
       (3, "Pling")
       (5, "Plang")
       (7, "Plong")
    ]
    |> List.map(p)
    |> String.concat ""
    |> function
       | "" -> sprintf "%d" number
       | plingPlangPlong -> plingPlangPlong

File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1]).

strIf(true, Str) -> Str;
strIf(false, _) -> "".

p(Number, Modulus, Str) -> 
	IsFactor = 0 =:= Number rem Modulus,
	strIf(IsFactor, Str).

num(Number, "") -> integer_to_list(Number);
num(_, PlingPlangPlong) -> PlingPlangPlong.

% num2(Number, PlingPlangPlong) ->
% 	case PlingPlangPlong of
% 		"" -> integer_to_list(Number);
% 		_ -> PlingPlangPlong
% 	end.

convert(Number) ->
	PlingPlangPlong = p(Number, 3, "Pling") ++ p(Number, 5, "Plang") ++ p(Number, 7, "Plong"),
	num(Number, PlingPlangPlong).
